felix/bpf/libbpf/libbpf_api.h

```diff
@@ -134,6 +134,43 @@ void bpf_tc_set_globals(struct bpf_map *map,
 	set_errno(bpf_map__set_initial_value(map, (void*)(&data), sizeof(data)));
 }
 
+int bpf_xdp_program_id(int ifIndex) {
+	__u32 prog_id = 0, flags = 0;
+	int err;
+
+	err = bpf_get_link_xdp_id(ifIndex, &prog_id, flags);
+	set_errno(err);
+	return prog_id;
+}
+
+int bpf_program_attach_xdp(struct bpf_object *obj, char *secName, int ifIndex, int old_id, __u32 flags)
+{
+	int err = 0;
+	struct bpf_link *link = NULL;
+	struct bpf_program *prog, *first_prog = NULL;
+	DECLARE_LIBBPF_OPTS(bpf_xdp_set_link_opts, opts,
+		.old_fd = bpf_prog_get_fd_by_id(old_id));
+
+	if (!(prog = bpf_object__find_program_by_title(obj, secName))) {
+		err = ENOENT;
+		goto out;
+	}
+
+	int prog_fd = bpf_program__fd(bpf_object__find_program_by_title(obj, secName));
```

Comment:
Please use ```bpf_program__fd(prog)``` instead of making another call to ```bpf_object__find_program_by_title```. Further I think maybe using ```bpf_object__find_program_by_title``` might be the reason for defining section name as ```xdp/<section_name>```.

Reply:
fixed it.
But the section is still the same.